cmd/registry/controller/controller.go

```diff
@@ -144,107 +136,246 @@ func generateActions(
+	// 1. Target resource is an api "projects/demo/locations/global/apis/petstore"
+	//    extracted parent will be "projects/demo/locations/global"
+	if strings.HasSuffix(parentName, "locations/global") {
+		// Return if this parent was already visited.
+		if visited[parentName] {
+			return nil, nil //[]*Action{}, nil
+		}
+
+		// Since the GeneratedResource is non-existent here,
+		// we will have to derive the exact name of the target resource.
+		targetResourceName, err := resourceNameFromParent(resourcePattern, parentName)
+		if err != nil {
+			return nil, fmt.Errorf("Cannot generate target resourceName to be created. Error: %s", err)
 		}
+
+		takeAction, err := needsCreate(
+			targetResourceName,
+			dependencyMaps,
+			generatedResource,
+		)
+
+		if err != nil {
+			return nil, err
+		} else if !takeAction {
+			return nil, nil //[]*Action{}, nil
+		}
+
+		cmd, err := generateCommand(generatedResource.Action, targetResourceName.String())
+		if err != nil {
+			return nil, fmt.Errorf("Cannot generate command: %s", err)
+		}
+
+		return []*Action{
+			{
+				Command:           cmd,
+				GeneratedResource: targetResourceName.String(),
+				RequiresReceipt:   generatedResource.Receipt,
+			},
+		}, nil
+	}
+
+	// If parent resource is not a project, then go through all the non-visited parents.
+	// We don't pass the filter here because the filter is for the target resource and not it's parent.
+	parentList, err := ListResources(ctx, client, parentName, "")
+	if err != nil {
+		return nil, err
+	}
+
+	actions := make([]*Action, 0)
+
+	for _, parent := range parentList {
+		// Skip if this parent was already visited.
+		if visited[parent.GetResourceName().String()] {
+			continue
+		}
+		// Since the GeneratedResource is non-existent here,
+		// we will have to derive the exact name of the target resource
+		targetResourceName, err := resourceNameFromParent(resourcePattern, parent.GetResourceName().String())
+		if err != nil {
+			log.FromContext(ctx).WithError(err).Error("Cannot generate target resourceName to be created.")
+			continue
+		}
+
+		takeAction, err := needsCreate(
+			targetResourceName,
+			dependencyMaps,
+			generatedResource,
+		)
+
+		if err != nil {
+			log.Errorf(ctx, "%s", err)
+			continue
+		} else if !takeAction {
+			continue
+		}
+
+		cmd, err := generateCommand(generatedResource.Action, targetResourceName.String())
+		if err != nil {
+			return nil, fmt.Errorf("Cannot generate command: %s", err)
+		}
+		a := &Action{
+			Command:           cmd,
+			GeneratedResource: targetResourceName.String(),
+			RequiresReceipt:   generatedResource.Receipt,
+		}
+		actions = append(actions, a)
 	}
 
 	return actions, nil
 }
+
+func needsUpdate(
+	targetResourceName ResourceName,
+	targetResourceTime time.Time,
+	dependencyMaps []map[string]time.Time,
+	generatedResource *rpc.GeneratedResource,
+	createMode bool) (bool, error) {
+	//Check if all the dependencies exist in the map.
```

Comment:
The loop does more than just this. IMO we should remove this (and the similar one in `needsCreate`) unless there's something to say that isn't clear from the loop code/comments.